app/services/migration/migrator.rb

```diff
@@ -59,5 +63,61 @@ def migrate_lead_providers
     def ecf_npq_lead_providers
       @ecf_npq_lead_providers ||= Ecf::NpqLeadProvider.all
     end
+
+    def ecf_cohorts
+      @ecf_cohorts ||= Ecf::Cohort.all
+    end
+
+    def migrate_cohorts
+      data_migration = DataMigration.find_by(model: :cohort)
+
+      data_migration.update!(started_at: Time.zone.now, total_count: ecf_cohorts.count)
```

Comment:
I feel like we could extract a lot of this to another method that accepts a block now its being reused, along the lines of:

```
def migrate(items, model)
  data_migration = DataMigration.find_by(model: :cohort)
  data_migration.update!(started_at: Time.zone.now, total_count: items.count)

  items.each do |item|  
    yield(item, data_migration)
    data_migration.increment!(:processed_count)
  end

  data_migration.update!(completed_at: Time.zone.now)
end
```

Then this method becomes something like:

```
def migrate_cohorts
  migrate(items, model) do
      cohort = Cohort.find_or_initialize_by(start_year: ecf_cohort.start_year)
      result = cohort.update(registration_start_date: ecf_cohort.registration_start_date)
      data_migration.increment!(:failure_count) unless result
  end
end
```